keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -310,7 +316,7 @@ class ClusterHandler(
     }
 
   override suspend fun export(exportable: Exportable): ClusterSpec {
-    // Get existing infrastructure
+    // Get existing infrastructure -- this is a very costly call
```

Comment:
I added this note because it routinely takes more than 30 seconds to retrieve the details for a single server group from clouddriver, which is insane... I'm not sure we could do much from our end, since it's not really information we would want to cache... Or maybe we do?

Reply:
Good question... One of my goals for the week is to take some time to teach myself metrics and Atlas (for real this time, unlike during onboarding 😝), so I could definitely add one here.

Reply:
> it routinely takes more than 30 seconds to retrieve the details for a single server group from clouddriver

wut? unless we're talking like p99 latency, that seems unbelievably wrong and bad. do you have a sense of what percentile 'routinely' maps to, and if it's not something crazy high like p99 have you considered reaching out to our platform team folks to figure out why we're seeing such unreasonable latency on a regular basis?

Reply:
Routinely as in > 90% of the calls. I believe I did reach out in chat about this at one point, and meant to do it again when I was testing this, but ended up forgetting. Will do.

Reply:
Update: collected some data from the logs and it turns out it's not a single call, but a combination of dozens of calls with near-second response times. But it looks like we can improve on our end, so digging into that now.

Reply:
@luispollo would you mind creating a separate issue to look into why this is taking 30+ seconds? You can assign it to me - I'm happy to look into it. I also feel it shouldn't take that long, maybe I can make it better.

Reply:
Here it is: https://github.com/spinnaker/keel/issues/1060
I'm already on it: https://github.com/spinnaker/clouddriver/pull/4540

Reply:
Worth removing this comment if you've fixed the problem? or do we still need to see if this fixes the problem?

Reply:
Still testing. Will remove the comment anyway as it's not actionable.